### ISSUE TITLE
no-extra-parens never reports double parens

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12127]
**Tell us about your environment**

* **ESLint Version:** 6.2.0
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

same in ESLint 5.16.0, this is not regression.

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2015
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is not reported because of the precedence:

```js
/*eslint no-extra-parens: "error"*/

a * ((b + c));
```

This is not reported because of the option (does the option allow multiple parens or just one pair?):

```js
/* eslint no-extra-parens: ["error", "all", { "conditionalAssign": false }] */

if (((a = b))) {}
```

There are many examples like this, seems that double parens are never reported whenever a single pair wouldn't be.

```bash
eslint index.js
```

**What did you expect to happen?**

Errors in both cases, I guess.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.

**Are you willing to submit a pull request to fix this bug?**

Yes, if this is indeed a bug. But this needs some analysis and design first, I'm thinking about an additional selector that would catch all double parens regardless of the context.

**Comment by platinumazure** [comment]
CC @mysticatea 

I'm wondering if this was caused by fee6acbe13cecd4c028e681e185fc6a6d6ba9452. I did remember seeing some changes in `no-extra-parens` in that commit...

**Comment by platinumazure** [comment]
@mdjermanovic Thanks for reporting. Any chance you can test with ESLint 6.1.x to make sure the issue isn't present there?

**Comment by mdjermanovic** [comment]
No, it wasn't that commit (tested with 6.1.0 to be sure), that change to a smarter `isParenthesised` might actually help to solve this, but this probably still needs a lot of work and testing. It's quite possible that these examples didn't work from the start, I believe this isn't a regression.

**Comment by platinumazure** [comment]
@mdjermanovic Just to confirm, you're saying the issue you reported also is also reproducible in 6.1.0? Thanks!

**Comment by mdjermanovic** [comment]
> @mdjermanovic Just to confirm, you're saying the issue you reported also is also reproducible in 6.1.0? Thanks!

Yes, sorry about the confusing post. The same issue is also reproducible in 6.1.0

**Comment by platinumazure** [comment]
Thanks @mdjermanovic, I've removed the "patch candidate" label based on your last statement.

I'm not 100% sure what the intended design of the rule is here. I'm hoping someone else on @eslint/eslint-team might have an idea on how best to handle this issue.

**Comment by mdjermanovic** [comment]
If this is not the intended design, the problem is with conditionals like this, that doesn't account for double parens:

```js
if (hasExcessParens(node.right) && precedence(node.right) >= precedence(node)) {
       report(node.right);
}
```

This could be solved by adding `|| hasDoubleExcessParens()` everywhere, or perhaps with a 'big' selector that would report double parens for all nodes, I'm not sure which solution is better. There is also a question of options, do they allow just one pair or more.

**Comment by mysticatea** [comment]
I think that to handle this issue as a bug makes sense.

**Comment by kaicataldo** [comment]
Reopening and assigning to myself so that it doesn't get auto closed again.

**Comment by kaicataldo** [comment]
Whoops, thanks @platinumazure
